common/aicore/path_finding.h

```diff
@@ -1,13 +1,6 @@
-/**************************************************************************
- Copyright (c) 1996-2020 Freeciv21 and Freeciv contributors. This file is
- __    __          part of Freeciv21. Freeciv21 is free software: you can
-/ \\..// \    redistribute it and/or modify it under the terms of the GNU
-  ( oo )        General Public License  as published by the Free Software
-   \__/         Foundation, either version 3 of the License,  or (at your
-                      option) any later version. You should have received
-    a copy of the GNU General Public License along with Freeciv21. If not,
-                  see https://www.gnu.org/licenses/.
-**************************************************************************/
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: Freeciv21 and Freeciv Contributors
+
 #pragma once
```

Comment:
This file needs `fc_types.h` for `enum direction8`, `vector` for `std::vector`, `unittype.h` for `bv_unit_types`, `tile.h` for `enum known_type`, and `support.h` for `fc__warn_unused_result`. After this `pf_tools.h` should be self-contained and clear to include first.

Reply:
Did you run a command of some kind to help you figure this out?

Reply:
I opened `pf_tools.h`, which revealed this:
![image](https://github.com/user-attachments/assets/7df86dd4-49d4-4f55-9856-588fa4540265)

So I opened that header and scrolled down to find the errors:
![image](https://github.com/user-attachments/assets/edae5759-046b-4f36-8484-6421e04865c5)

All in Kate with the `clangd` diagnostics enabled.

Reply:
`clang-tidy-17 -p build-clang common/aicore/path_finding.h` gives the same